api/src/main/java/com/community/api/endpoint/avisoft/controller/Account/AccountEndPoint.java

```diff
@@ -277,18 +278,60 @@ private ResponseEntity<?> loginWithPhoneOtp(Map<String, Object> loginDetails, Ht
                     return responseService.generateErrorResponse(ApiConstants.NO_RECORDS_FOUND, HttpStatus.NOT_FOUND);
                 }
 
-            } else if (roleService.findRoleName(role).equals(Constant.ADMIN) || roleService.findRoleName(role).equals(Constant.SUPER_ADMIN) || roleService.findRoleName(role).equals(Constant.roleAdminServiceProvider)) {
-                if (adminService.findAdminByPhone(mobileNumber, countryCode) != null) {
-                    if (adminService.findAdminByPhone(mobileNumber, countryCode).getOtp() != null) {
-                        responseService.generateErrorResponse(ApiConstants.NO_RECORDS_FOUND, HttpStatus.NOT_FOUND);
+            }
+            else if (roleService.findRoleName(role).equals(Constant.ADMIN) ) {
+                CustomAdmin customAdmin = adminService.findAdminByPhone(mobileNumber,countryCode);
+                if (customAdmin != null) {
+                    if (customAdmin.getRole() == 2) {
+                        if (adminService.findAdminByPhone(mobileNumber, countryCode).getOtp() != null) {
+                            responseService.generateErrorResponse(ApiConstants.NO_RECORDS_FOUND, HttpStatus.NOT_FOUND);
+                        }
+                        return adminService.sendOtpForAdmin(mobileNumber, countryCode, session);
                     }
-                    return adminService.sendOtpForAdmin(mobileNumber, countryCode, session);
-                } else {
+                    else{
+                        return responseService.generateErrorResponse("Custom Admin with mobileNumber " + mobileNumber + " does not have "+ roleService.findRoleName(role)+" role", HttpStatus.BAD_REQUEST);
+                    }
+                }
+                else {
                     return responseService.generateErrorResponse(ApiConstants.NO_RECORDS_FOUND, HttpStatus.NOT_FOUND);
                 }
-            } else {
+            }
+            else if (roleService.findRoleName(role).equals(Constant.SUPER_ADMIN) ) {
+                CustomAdmin customAdmin = adminService.findAdminByPhone(mobileNumber,countryCode);
+                if (customAdmin != null) {
+                    if (customAdmin.getRole() == 1) {
```

Comment:
To write the logic more clearly and efficiently, we can define constants for role names and then retrieve the role using the roleService.findRoleById(roleId) method. After obtaining the role object, we can compare the role name using equals() to check if it matches the expected constant.